overlay.py

```diff
@@ -50,6 +50,7 @@ def __init__(self, bike, width=1280, height=740, bg: str = None):
         self.client.on_connect = self._on_connect
         self.client.on_disconnect = self.on_disconnect
         self.client.on_log = self.on_log
+        self.client.is_connected = False  # Default is false until connected
```

Comment:
It's best that we don't write new attributes to objects like this, especially when they're not one we've made and understand the internals of! If we were to go with keeping track of whether we're connected on our own, then we'd be best to stick with our own separate variable. However, lucky for us, the `paho-mqtt` client can tell us on it's own whether it's connected!

It's not listed in the docs for some reason, but in the source code [here](https://github.com/eclipse/paho.mqtt.python/blob/d7592a64b55469035ab66160919561c30ae5f645/src/paho/mqtt/client.py#L1349) we can see there's a public `is_connected()` method we can use to check for connection instead. In fact, this method would have been replaced by your boolean on this line! This is one reason why we shouldn't write new attributes to objects like this :)

Reply:
Didn't realise this existed, thanks for sharing!